Get rid of the old Singleton class

Appled opened the pull request:
Get rid of the old Singleton class and replace it with modern and thread-safe c++11 static method. 
This issue finally fixes https://github.com/AscEmu/AscEmu/issues/574 and AE can be run on Linux 🎉 

Assertion failed in the Singleton class (in the issue) because Singleton "rules" have been constantly violated. Almost every Singleton child class had "new" methods called in both world and logonserver startup. As well in the shutdown methods for both world and logonserver "delete" or destructor methods are called.

Therefore I:
- have made constructors for each class private and defaulting them
- have implemented "void initialize()" function to these classes which needed an initializition before use
- have implemented "::getInstance()" to these classes which will return a thread-safe static variable (replaces old "::getSingleton()")
- have rewritten some of the #defines for better clarify, for example: bgMgr is sBattlegroundManager or objmgr is sObjectMgr
- have removed Singleton.h

I didn't touch the destructors and I left them mostly as they were but in the future, since the Singleton object should last as long as the program does, classes should be finalized and destructors should be made private, and instead use a function specifically made for closing log files etc. 

Closes https://github.com/AscEmu/AscEmu/issues/574

Zyres commented on src/world/Server/World.cpp:
```diff
@@ -792,13 +776,13 @@ bool World::setInitialWorldSettings()
 
-    LogDetail("World : Starting CBattlegroundManager...");
-    new CBattlegroundManager;
+    LogDetail("World : Starting CsBattlegroundManager...");
```
class renaming is not part of the PR

Appled commented:
Update: I have changed getInstance() to return a reference and I have also made destruction methods private. ::finalize() replaces some old destruction methods.
In addition I have deleted the copy and assignment operators.
